client/src/main/java/tds/assessment/Assessment.java

```diff
@@ -1,40 +1,50 @@
 package tds.assessment;
 
+import java.util.List;
+
 /**
- * Represents admin information concerning an assessment
+ * Represents admin information concerning an assessment and its individual segments (if any)
  */
-public class SetOfAdminSubject {
+public class Assessment {
     private String key;
     private String assessmentId;
-    private boolean segmented;
     private String selectionAlgorithm;
     private float startAbility;
     private String subjectName;
+    private List<Segment> segments;
 
-    public SetOfAdminSubject(String key,
-                             String assessmentId,
-                             boolean segmented,
-                             String selectionAlgorithm,
-                             float startAbility) {
-        this(key, assessmentId, segmented, selectionAlgorithm, startAbility, null);
+    public Assessment(String key,
+                      String assessmentId,
+                      List<Segment> segments,
+                      String selectionAlgorithm,
+                      float startAbility) {
+        this(key, assessmentId, segments, selectionAlgorithm, startAbility, null);
     }
 
-    public SetOfAdminSubject(String key,
-                             String assessmentId,
-                             boolean segmented,
-                             String selectionAlgorithm,
-                             float startAbility,
-                             String subjectName) {
+    public Assessment(String key,
+                      String assessmentId,
+                      List<Segment> segments,
+                      String selectionAlgorithm,
+                      float startAbility,
+                      String subjectName) {
         this.key = key;
         this.assessmentId = assessmentId;
-        this.segmented = segmented;
+        this.segments = segments;
```

Comment:
What if we created a [umodifiableList](https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#unmodifiableList(java.util.List)) since the list should never change once created?